src/api/custom-resolvers/mutations/account-sign-in/account-sign-in-checks/index.test.ts

```diff
@@ -55,39 +54,115 @@ describe('custom-resolvers/account-sign-in/account-sign-in-checks', () => {
 
     accessCodeEmailSpy = jest.fn(() => Promise.resolve(mockSendEmailResponse));
     sendEmail.accessCodeEmail = accessCodeEmailSpy;
-
-    result = await accountSignInChecks(context, account, mockUrlOrigin);
-
-    account = await accounts.get(context, account.id);
   });
 
   describe('when the account is verified', () => {
-    test('it should generate an OTP and save to the account', () => {
+    it('should generate an OTP and save to the account', async () => {
+      await accountSignInChecks(context, account, mockUrlOrigin);
+
+      account = await accounts.get(context, account.id);
+
       expect(account.otpSalt).toEqual(mockOTP.salt);
       expect(account.otpHash).toEqual(mockOTP.hash);
       expect(new Date(account.otpExpiry)).toEqual(mockOTP.expiry);
     });
 
-    test('it should call sendEmail.accessCodeEmail', () => {
-      const { email } = account;
+    describe('when NODE_ENV is `development`', () => {
+      beforeAll(() => {
+        process.env.NODE_ENV = 'development';
+      });
+
+      beforeEach(() => {
+        jest.resetAllMocks();
+
+        console.info = consoleSpy;
+
+        accessCodeEmailSpy = jest.fn(() => Promise.resolve(mockSendEmailResponse));
+        sendEmail.accessCodeEmail = accessCodeEmailSpy;
+      });
 
-      const name = getFullNameString(account);
+      it('should NOT call sendEmail.confirmEmailAddress', async () => {
+        await accountSignInChecks(context, account, mockUrlOrigin);
+
+        expect(accessCodeEmailSpy).toHaveBeenCalledTimes(0);
+      });
 
-      expect(accessCodeEmailSpy).toHaveBeenCalledTimes(1);
-      expect(accessCodeEmailSpy).toHaveBeenCalledWith(email, name, mockOTP.securityCode);
+      it('should log out an OTP', async () => {
+        await accountSignInChecks(context, account, mockUrlOrigin);
+
+        const flattenedSpys = consoleSpy.mock.calls.flat();
+
+        const assertion1 = flattenedSpys.includes('✅ Signing in account (dev environment only) - mimicking sending OTP via email %s');
+        const assertion2 = flattenedSpys.includes(mockOTP.securityCode);
+
+        expect(assertion1).toEqual(true);
+        expect(assertion2).toEqual(true);
+      });
+
+      it('should return an accountId and success=true', async () => {
+        const result = await accountSignInChecks(context, account, mockUrlOrigin);
+
+        const expected = {
+          accountId: account.id,
+          success: true,
+        };
+
+        expect(result).toEqual(expected);
+      });
     });
 
-    test('it should return the email response and accountId', () => {
-      const expected = {
-        ...mockSendEmailResponse,
-        accountId: account.id,
-      };
+    describe('when NODE_ENV is NOT `development`', () => {
+      consoleSpy = jest.fn();
 
-      expect(result).toEqual(expected);
+      beforeAll(() => {
+        process.env.NODE_ENV = 'not-development';
```

Comment:
does this need to be reset after the test has run?

Reply:
Whoops, forgot to add in this test. Updated `afterAll`.